app.ts

```diff
@@ -0,0 +1,163 @@
+﻿const host = "http://localhost:2050";
+let grid: Grid;
+const initRecord = 0;
+const rowLimit = 22;
+
+class Grid {
+    from: number;
+    to: number;
+    headers: string[] = [];
+
+    $row = $("<div />", {
+        class: 'row'
+    });
+
+    $header = $("<div />", {
+        class: 'header'
+    });
+
+    $column = $("<div />", {
+        class: 'column'
+    });
+
+    //sets initial grid values.
+    constructor(from: number, to: number) {
+        this.from = from;
+        this.to = to;
+
+        $.getJSON(host + "/columns", (data) => { })
+            .done((headers) => {
+                this.headers = headers;
+                this.pageInit();
+
+                //adds initial records to row objects to be added to grid on page load.
+                $.getJSON(host + "/records?from=" + this.from + "&to=" + this.to, (data) => { })
+                    .done((records) => {
+                        this.drawGrid(records);
+                    });
+            });
+    }
+
+    //(re)initializes the page deafault values
+    pageInit() {
+        $('#grid').empty();
+        this.$row.empty();
+        this.$column.empty();
+
+        //add headers to row object to be added to the grid on initialization.
+        for (let col of this.headers) {
+            this.$row.append(this.$header.clone().text(col));
+        }
+
+        $("#grid").append(this.$row.clone());
+    }
+
+    //iterates through records and adds them as rows to the grid
+    drawGrid(records: string[][]) {
+        for (let record of records) {
+            this.$row.empty();
+
+            for (let col of record) {
+                this.$row.append(this.$column.clone().text(col));
+            }
+
+            $("#grid").append(this.$row.clone());
+        }
+    }
+
+    //loads the higher id value set of records, than the current set of records, to display on the grid.
+    nextPage() {
+        this.pageInit();
+        this.from = this.to + 1;
+        this.to = this.to + rowLimit + 1;
+
+        $.getJSON(host + "/recordCount", (result) => {
```

Comment:
I wanted to load it to its own var but figured in a real world setting where the recordCount might have changed since the last click, it would be better to check it again. but in this case ...  no its not